pyterrier/io.py

```diff
@@ -212,22 +213,23 @@ def write_results(res, filename, format="trec", **kwargs):
         raise ValueError("Format %s not known, supported types are %s" % (format, str(SUPPORTED_RESULTS_FORMATS.keys())))
     # convert generators to results 
     res = coerce_dataframe(res)
-    return SUPPORTED_RESULTS_FORMATS[format][1](res, filename, **kwargs)
+    return SUPPORTED_RESULTS_FORMATS[format][1](res, filename, append=append, **kwargs)
 
-def _write_results_trec(res, filename, run_name="pyterrier"):
+def _write_results_trec(res, filename, run_name="pyterrier", append=False):
         res_copy = res.copy()[["qid", "docno", "rank", "score"]]
         res_copy.insert(1, "Q0", "Q0")
         res_copy.insert(5, "run_name", run_name)
-        res_copy.to_csv(filename, sep=" ", header=False, index=False)
+        res_copy.to_csv(filename, sep=" ", mode='wa' if append else 'w', header=False, index=False)
```

Comment:
I think these need to just be `'a'` (instead of `'wa'`)

Reply:
Default mode for pandas is 'w', so the documentation said

Reply:
My point is that `'wa'` isn't a valid file mode. E.g.,

```python
pd.DataFrame().to_csv('test.csv', mode='wa')
# ValueError: must have exactly one of create/read/write/append mode
```